src/neo-vm/ExecutionEngine.cs

```diff
@@ -1186,13 +1345,29 @@ internal protected void ExecuteNext()
                 {
                     Instruction instruction = CurrentContext.CurrentInstruction;
                     if (!PreExecuteInstruction() || !ExecuteInstruction() || !PostExecuteInstruction(instruction))
+                    {
                         State = VMState.FAULT;
+                        FaultState.HoldError = false;
+                        FaultState.ErrorInfo = "OPCode Fault:" + instruction.OpCode.ToString();
+                    }
+                }
+                catch (InvalidProgramException)
+                {
+                    FaultState.ErrorInfo = "Can't catch internal error:";
+                    FaultState.HoldError = false;
+                    State = VMState.FAULT;
```

Comment:
@Tommo-L we should remove the `ErrorHandler` ? whats happen if we call Execute again with a FAULT?

Reply:
HandleError(); has one chance to fix the fault State.

Reply:
when fault,will try to handle error once on

Reply:
@shargon I removed the `return` in catch-InvalidProgramException

```c#
try
                {
                    Instruction instruction = CurrentContext.CurrentInstruction;
                    if (!PreExecuteInstruction() || !ExecuteInstruction() || !PostExecuteInstruction(instruction))
                    {
                        State = VMState.FAULT;
                        FaultState.HoldError = false;
                        FaultState.ErrorInfo = "OPCode Fault:" + instruction.OpCode.ToString();
                    }
                }
                catch (InvalidProgramException)
                {
                    FaultState.ErrorInfo = "Can't catch internal error:";
                    FaultState.HoldError = false;
                    State = VMState.FAULT;
                    return; // <---- removed
                }
                catch
                {
                    State = VMState.FAULT;
                    FaultState.HoldError = false;
                }
```

Reply:
We should decide which should throw `InvalidProgramException`

Reply:
Yes
For me, I think the `try-catch-finally` only handle the error throwed by `throw/throwif/throwifnot`. For the other exceptions that we should set vm state = FAULT.

Reply:
for me，try catch should handle every error，we will add errortype in future，let catch(type) decide which error will be catch